arpack fails to build under aarch64 Darwin

### Describe the bug

Reported via email by a Nix user without a GitHub account.

### Steps To Reproduce

```console
$ nix-build -A arpack
building '/nix/store/pb2102yd1x50ls4592zi48jjirsddmwc-arpack-3.8.0.drv'...
unpacking sources
unpacking source archive /nix/store/vn4vxkl4c218smc8k8yxcww77n9ssl8x-source
source root is source
patching sources
applying patch /nix/store/9apqzvzdlwwy5p72ra1r9a1aswqjci71-pkg-config-paths.patch
patching file CMakeLists.txt
updateAutotoolsGnuConfigScriptsPhase
configuring
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DCMAKE_INSTALL_LOCALEDIR=
/nix/store/a9qs6hnzwrhgvqhc42762dx25ky35aav-arpack-3.8.0/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/a9qs6hnzwrhgvqhc42762dx25ky35aav-arpack-3.8.0/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/a9qs6hnzwrhgvqhc427
62dx25ky35aav-arpack-3.8.0/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/a9qs6hnzwrhgvqhc42762dx25ky35aav-arpack-3.8.0/share/doc/arpack -DCMAKE_INSTALL_INFODIR=/nix/store/a9qs6hnzwrhgvqhc42762dx25ky35aav-arpack-3.8.0/share/
info -DCMAKE_INSTALL_MANDIR=/nix/store/a9qs6hnzwrhgvqhc42762dx25ky35aav-arpack-3.8.0/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/a9qs6hnzwrhgvqhc42762dx25ky35aav-arpack-3.8.0/include -DCMAKE_INSTALL_INCLUDEDI
R=/nix/store/a9qs6hnzwrhgvqhc42762dx25ky35aav-arpack-3.8.0/include -DCMAKE_INSTALL_SBINDIR=/nix/store/a9qs6hnzwrhgvqhc42762dx25ky35aav-arpack-3.8.0/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/a9qs6hnzwrhgvqhc42762dx25ky3
5aav-arpack-3.8.0/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/a9qs6hnzwrhgvqhc42762dx25ky35aav-arpack-3.8.0/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/stor
e/jgaarpq3rdfk7gbm0gm3xlh8hhbw15ay-clang-wrapper-11.1.0/bin/strip -DCMAKE_RANLIB=/nix/store/7lfjbip2dvhvnhvdj0kx4d63apqlz83v-cctools-binutils-darwin-949.0.1/bin/ranlib -DCMAKE_AR=/nix/store/7lfjbip2dvhvnhvdj0kx4d63a
pqlz83v-cctools-binutils-darwin-949.0.1/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/a9qs6hnzwrhgvqhc42762dx25ky35aav-arpack-3.8.0 -DBUILD_SHARED_LIBS=ON -DINTERFACE
64=0
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
Compatibility with CMake < 2.8.12 will be removed from a future version of
CMake.
Update the VERSION argument <min> value or use a ...<max> suffix to tell
CMake that the project does not need compatibility with older versions.
-- The C compiler identification is Clang 11.1.0
-- The Fortran compiler identification is unknown
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/jgaarpq3rdfk7gbm0gm3xlh8hhbw15ay-clang-wrapper-11.1.0/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Checking whether Fortran compiler has -isysroot
-- Checking whether Fortran compiler has -isysroot - yes
-- Checking whether Fortran compiler supports OSX deployment target flag
-- Checking whether Fortran compiler supports OSX deployment target flag - yes
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - failed
-- Check for working Fortran compiler: /nix/store/08nypsgnh2skq55xxmv38s45hffj99wn-gfortran-wrapper-11.1.0/bin/gfortran
-- Check for working Fortran compiler: /nix/store/08nypsgnh2skq55xxmv38s45hffj99wn-gfortran-wrapper-11.1.0/bin/gfortran - broken
CMake Error at /nix/store/zpf2rni9fdrx5flx5xab6wjnb16jifr6-cmake-3.21.2/share/cmake-3.21/Modules/CMakeTestFortranCompiler.cmake:54 (message):
The Fortran compiler
"/nix/store/08nypsgnh2skq55xxmv38s45hffj99wn-gfortran-wrapper-11.1.0/bin/gfortran"
is not able to compile a simple test program.
It fails with the following output:
Change Dir: /tmp/nix-build-arpack-3.8.0.drv-0/source/build/CMakeFiles/CMakeTmp
Run Build Command(s):/nix/store/57dwqr6igadm896armq61h96psdni160-gnumake-4.3/bin/make -f Makefile cmTC_3c7b7/fast && /nix/store/57dwqr6igadm896armq61h96psdni160-gnumake-4.3/bin/make -f CMakeFiles/cmTC_3c7b7.dir
/build.make CMakeFiles/cmTC_3c7b7.dir/build
make[1]: Entering directory '/private/tmp/nix-build-arpack-3.8.0.drv-0/source/build/CMakeFiles/CMakeTmp'
Building Fortran object CMakeFiles/cmTC_3c7b7.dir/testFortranCompiler.f.o
/nix/store/08nypsgnh2skq55xxmv38s45hffj99wn-gfortran-wrapper-11.1.0/bin/gfortran
-c /tmp/nix-build-arpack-3.8.0.drv-0/source/build/CMakeFiles/CMakeTmp/testFortranCompiler.f -o CMakeFiles/cmTC_3c7b7.dir/testFo
rtranCompiler.f.o
f951: Warning: command-line option '-iframework /nix/store/ns2z57b334ylk9dn12f7n7wfn9gd3fh6-apple-framework-CoreFoundation-11.0.0/Library/Frameworks' is valid for C/C++/ObjC/ObjC++ but not for Fortran
f951: Warning: command-line option '-iframework /nix/store/ns2z57b334ylk9dn12f7n7wfn9gd3fh6-apple-framework-CoreFoundation-11.0.0/Library/Frameworks' is valid for C/C++/ObjC/ObjC++ but not for Fortran
f951: Warning: command-line option '-Wformat=1' is valid for C/C++/ObjC/ObjC++ but not for Fortran
f951: Warning: command-line option '-Wformat-security' is valid for C/C++/ObjC/ObjC++ but not for Fortran
f951: Warning: '-Werror=' argument '-Werror=format-security' is not valid for Fortran
/private/tmp/nix-build-arpack-3.8.0.drv-0/ccPeIBZ8.s:24:15: error: index must be an integer in range [-256, 255].
ldr
x2, [x2, ___stack_chk_guard];momd
^
/private/tmp/nix-build-arpack-3.8.0.drv-0/ccPeIBZ8.s:46:15: error: index must be an integer in range [-256, 255].
ldr
x0, [x0, ___stack_chk_guard];momd
^
make[1]: *** [CMakeFiles/cmTC_3c7b7.dir/build.make:78: CMakeFiles/cmTC_3c7b7.dir/testFortranCompiler.f.o] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-arpack-3.8.0.drv-0/source/build/CMakeFiles/CMakeTmp'
make: *** [Makefile:127: cmTC_3c7b7/fast] Error 2
CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
CMakeLists.txt:7 (project)
-- Configuring incomplete, errors occurred!
See also "/tmp/nix-build-arpack-3.8.0.drv-0/source/build/CMakeFiles/CMakeOutput.log".
See also "/tmp/nix-build-arpack-3.8.0.drv-0/source/build/CMakeFiles/CMakeError.log".
builder for '/nix/store/pb2102yd1x50ls4592zi48jjirsddmwc-arpack-3.8.0.drv' failed with exit code 1
```

### Expected behavior

Build fines.

### Additional context

In the attempt to build GNU octave for aarch64 darwin.

### Notify maintainers

@ttuegel @dotlambda

### Metadata

```
$ nix-info
- system: `"aarch64-darwin"`
- host os: `Darwin 20.6.0, macOS 11.6`
- multi-user?: `no`
- sandbox: `no`
- version: `nix-env (Nix) 2.3.15`
- channels(eegenolf): `"nixpkgs-21.11pre318361.a3a23d9599b"`
- nixpkgs: `/Users/eegenolf/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: arpack
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/libraries/science/math/arpack/default.nix
+++ b/pkgs/development/libraries/science/math/arpack/default.nix
@@ -49,6 +49,10 @@ stdenv.mkDerivation rec {
     install_name_tool -change libblas.dylib ${blas}/lib/libblas.dylib $out/lib/libarpack.dylib
   '';
 
+  # disable stackprotector on aarch64-darwin for now
+  # https://github.com/NixOS/nixpkgs/issues/127608
+  hardeningDisable = lib.optionals (stdenv.isAarch64 && stdenv.isDarwin) [ "stackprotector" ];
+
   meta = {
     homepage = "https://github.com/opencollab/arpack-ng";
     description = ''

